src/Aspire.Dashboard/Telemetry/DashboardTelemetryService.cs

```diff
@@ -0,0 +1,390 @@
+        });
+
+        return context;
+    }
+
+    /// <summary>
+    /// Post a session property.
+    /// </summary>
+    public void PostProperty(string propertyName, AspireTelemetryProperty propertyValue)
+    {
+        if (!IsEnabled())
+        {
+            return;
+        }
+
+        var context = OperationContext.Create(propertyCount: 0, name: TelemetryEndpoints.TelemetryPostProperty);
+        telemetrySender.QueueRequest(context, async (client, _) =>
+        {
+            var request = new PostPropertyRequest(propertyName, propertyValue);
+            await client.PostAsJsonAsync(TelemetryEndpoints.TelemetryPostProperty, request).ConfigureAwait(false);
+        });
+    }
+
+    /// <summary>
+    /// Post a session recurring property.
+    /// </summary>
+    public void PostRecurringProperty(string propertyName, AspireTelemetryProperty propertyValue)
+    {
+        if (!IsEnabled())
+        {
+            return;
+        }
+
+        var context = OperationContext.Create(propertyCount: 0, name: TelemetryEndpoints.TelemetryPostRecurringProperty);
+        telemetrySender.QueueRequest(context, async (client, _) =>
+        {
+            var request = new PostPropertyRequest(propertyName, propertyValue);
+            await client.PostAsJsonAsync(TelemetryEndpoints.TelemetryPostRecurringProperty, request).ConfigureAwait(false);
+        });
+    }
+
+    /// <summary>
+    /// Currently not used.
+    /// </summary>
+    public void PostCommandLineFlags(List<string> flagPrefixes, Dictionary<string, AspireTelemetryProperty> additionalProperties)
+    {
+        if (!IsEnabled())
+        {
+            return;
+        }
+
+        var context = OperationContext.Create(propertyCount: 0, name: TelemetryEndpoints.TelemetryPostCommandLineFlags);
+        telemetrySender.QueueRequest(context, async (client, _) =>
+        {
+            var request = new PostCommandLineFlagsRequest(flagPrefixes, additionalProperties);
+            await client.PostAsJsonAsync(TelemetryEndpoints.TelemetryPostCommandLineFlags, request).ConfigureAwait(false);
+        });
+    }
+
+    /// <summary>
+    /// Gets identifying properties for the telemetry session.
+    /// </summary>
+    public Dictionary<string, AspireTelemetryProperty> GetDefaultProperties()
+    {
+        return new Dictionary<string, AspireTelemetryProperty>
+        {
+            { TelemetryPropertyKeys.DashboardVersion, new AspireTelemetryProperty(typeof(DashboardWebApplication).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? string.Empty) },
+            { TelemetryPropertyKeys.DashboardBuildId, new AspireTelemetryProperty(typeof(DashboardWebApplication).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version ?? string.Empty) },
+        };
+    }
+
+    public ValueTask<string> GetUserAgentAsync(IJSRuntime js)
+    {
+        return js.InvokeAsync<string>("getUserAgent");
+    }
```

Comment:
The user agent is used with telemetry but this doesn't seem like the right type to place it on. I guess having it here gives a place to cache the value.